# Exploring Ebay Car Sales Data
## Project Purpose:
The project aims to clean the data and analyze the included used car listings.
## Data Source:
The dataset of used cars from eBay Kleinanzeigen was originally scraped and uploaded to Kaggle. 50000 data points from the full dataset were sampled to run the project.
## The data dictionary provided with data is as follows:
dateCrawled - When this ad was first crawled. All field-values are taken from this date.

name - Name of the car.

seller - Whether the seller is private or a dealer.

offerType - The type of listing

price - The price on the ad to sell the car.

abtest - Whether the listing is included in an A/B test.

vehicleType - The vehicle Type.

yearOfRegistration - The year in which the car was first registered.

gearbox - The transmission type.

powerPS - The power of the car in PS.

model - The car model name.

kilometer - How many kilometers the car has driven.

monthOfRegistration - The month in which the car was first registered.

fuelType - What type of fuel the car uses.

brand - The brand of the car.

notRepairedDamage - If the car has a damage which is not yet repaired.

dateCreated - The date on which the eBay listing was created.

nrOfPictures - The number of pictures in the ad.

postalCode - The postal code for the location of the vehicle.

lastSeenOnline - When the crawler saw this ad last online.

In [1]:
import pandas as pd
import numpy as np

In [2]:
autos = pd.read_csv('autos.csv', encoding = 'Latin-1' )

In [3]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

The dataset contains 20 columns, of which most are strings. Some columns contain NaN value. The column names use camelcase instead of Python's preferred snakecase.

In [4]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

We are going to rename some of the columns and change them from camelcase to snakecase.

In [5]:
column_rename = {'dateCrawled':'date_crawled','name':'name', 'seller':'seller', 'offerType':'offer_type', 'price':'price', 'abtest':'abtest',
       'vehicleType':'vehicle_type', 'yearOfRegistration':'registration_year', 'gearbox':'gearbox', 'powerPS':'power_PS', 'model':'model',
       'odometer':'odometer', 'monthOfRegistration':'registration_month', 'fuelType':'fuel_type', 'brand':'brand',
       'notRepairedDamage':'unrepaired_damage', 'dateCreated':'ad_created', 'nrOfPictures':'num_picture', 'postalCode':'postal_code',
       'lastSeen':'last_seen'}
autos.rename(column_rename, axis = 1, inplace = True)
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_PS   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  num_picture  \
0     peugeot              nein  2016-03-26 00:00:00            0   
1         bmw              nein  2016-04-04 00:00:00            0   
2  volkswagen              nein  2016-03-26 00:00:00            0   
3       smart              nein  2016-03-12 00:00:00            0   
4        ford              nein  2016-04-01 00:00:00            0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

In [6]:
autos.describe()

registration_year      power_PS  registration_month  num_picture  \
count       50000.000000  50000.000000        50000.000000      50000.0   
mean         2005.073280    116.355920            5.723360          0.0   
std           105.712813    209.216627            3.711984          0.0   
min          1000.000000      0.000000            0.000000          0.0   
25%          1999.000000     70.000000            3.000000          0.0   
50%          2003.000000    105.000000            6.000000          0.0   
75%          2008.000000    150.000000            9.000000          0.0   
max          9999.000000  17700.000000           12.000000          0.0   

        postal_code  
count  50000.000000  
mean   50813.627300  
std    25779.747957  
min     1067.000000  
25%    30451.000000  
50%    49577.000000  
75%    71540.000000  
max    99998.000000

Let's first look at the columns with numerical values.
- 'num_picture' has no other values but 0.

Therefore we can drop the 'num_picture' column.

In [7]:
autos = autos.drop('num_picture', axis = 1)

In [8]:
autos.describe(include = ['O'])

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-04-02 11:37:04  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   

       vehicle_type  gearbox  model   odometer fuel_type       brand  \
count         44905    47320  47242      50000     45518       50000   
unique            8        2    245         13         7          40   
top       limousine  manuell   golf  150,000km    benzin  volkswagen   
freq          12859    36993   4024      32424     30107       10687   

       unrepaired_damage           ad_created            last_seen  
count              40171                50000                50000  
unique                 2                   76                39481  
top                 nein  2016-04-03 00:00:00  2016-04-07 06:17:27  
freq               35232                 1946                    8

Let's then look at the columns with string values.
- 'seller' and 'offer_type' columns are near all values are same (49999 out of 50000).

Therefore, we could drop these two columns.

In [9]:
autos = autos.drop(['seller','offer_type'],axis = 1)

We also notice that the columns 'price' and 'odometer' are text values. It should be numeric.
<br>
First we need to have a look at the two columns.

In [10]:
autos['price'].value_counts()

$0         1421
$500        781
$1,500      734
$2,500      643
$1,000      639
           ... 
$6,530        1
$1,039        1
$11,780       1
$45,800       1
$4,895        1
Name: price, Length: 2357, dtype: int64

In [11]:
autos['odometer'].value_counts()

150,000km    32424
125,000km     5170
100,000km     2169
90,000km      1757
80,000km      1436
70,000km      1230
60,000km      1164
50,000km      1027
5,000km        967
40,000km       819
30,000km       789
20,000km       784
10,000km       264
Name: odometer, dtype: int64

Let's replace the string values of these two columns to numetical values and change the column names.

In [12]:
autos['price'] = (autos['price']
                  .str.replace('$','')
                  .str.replace(',','')
                  .astype(int)
                 )
autos['odometer'] = (autos['odometer']
                     .str.replace('km','')
                     .str.replace(',','')
                     .astype(int)
                    )
autos.rename({'price':"price_$",'odometer':'odometer_km'}, axis = 1, inplace = True)

Let's have a look at the dataset infomation after cleaning.

In [13]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 17 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   date_crawled        50000 non-null  object
 1   name                50000 non-null  object
 2   price_$             50000 non-null  int32 
 3   abtest              50000 non-null  object
 4   vehicle_type        44905 non-null  object
 5   registration_year   50000 non-null  int64 
 6   gearbox             47320 non-null  object
 7   power_PS            50000 non-null  int64 
 8   model               47242 non-null  object
 9   odometer_km         50000 non-null  int32 
 10  registration_month  50000 non-null  int64 
 11  fuel_type           45518 non-null  object
 12  brand               50000 non-null  object
 13  unrepaired_damage   40171 non-null  object
 14  ad_created          50000 non-null  object
 15  postal_code         50000 non-null  int64 
 16  last_seen           50

Let's first analyze the 'price_$' column.

In [14]:
autos['price_$'].unique().shape

(2357,)

We know there are 2357 different prices.

In [15]:
autos['price_$'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price_$, dtype: float64

In [16]:
autos['price_$'].value_counts().sort_index(ascending = True).head(20)

0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
13       2
14       1
15       2
17       3
18       1
20       4
25       5
29       1
30       7
35       1
Name: price_$, dtype: int64

There are 1592 rows with value less than $10, which is not reasonable for the price of a car.
<br>
Therefore, we could delete these rows.

In [17]:
autos['price_$'].value_counts().sort_index(ascending = False).head(20)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
Name: price_$, dtype: int64

There are 8 rows with values equal or larger than 10000000, which is not reasonable for the price of a car.
<br>
Therefore, we could delete these rows.

In [18]:
autos = autos[(autos['price_$'] > 10) & (autos['price_$'] < 10000000)]
autos['price_$'].describe()

count    4.840000e+04
mean     6.103728e+03
std      2.282433e+04
min      1.100000e+01
25%      1.250000e+03
50%      3.000000e+03
75%      7.499000e+03
max      3.890000e+06
Name: price_$, dtype: float64

Let's then analyze the 'odometer_km' column.

In [19]:
autos['odometer_km'].unique().shape

(13,)

We know there are 13 different values in the 'odometer_km' column.

In [20]:
autos['odometer_km'].describe()

count     48400.000000
mean     125802.376033
std       39716.971871
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [21]:
autos['odometer_km'].value_counts()

150000    31303
125000     5047
100000     2108
90000      1733
80000      1414
70000      1215
60000      1154
50000      1012
40000       815
5000        810
30000       779
20000       761
10000       249
Name: odometer_km, dtype: int64

We notice there are more cars with higher mileage than those with lower milegae.

Let's then look at the date columns: 'date_crawled', 'last_seen', 'ad_created', 'registration_month', 'registration_year'.

In [22]:
autos[['date_crawled','last_seen','ad_created','registration_month','registration_year']].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48400 entries, 0 to 49999
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   date_crawled        48400 non-null  object
 1   last_seen           48400 non-null  object
 2   ad_created          48400 non-null  object
 3   registration_month  48400 non-null  int64 
 4   registration_year   48400 non-null  int64 
dtypes: int64(2), object(3)
memory usage: 2.2+ MB


Among them, 'date_crawled', 'last_seen' and 'ad_created' are identifed as string values.

In [23]:
autos[['date_crawled','last_seen','ad_created']].head(10)

date_crawled            last_seen           ad_created
0  2016-03-26 17:47:46  2016-04-06 06:45:54  2016-03-26 00:00:00
1  2016-04-04 13:38:56  2016-04-06 14:45:08  2016-04-04 00:00:00
2  2016-03-26 18:57:24  2016-04-06 20:15:37  2016-03-26 00:00:00
3  2016-03-12 16:58:10  2016-03-15 03:16:28  2016-03-12 00:00:00
4  2016-04-01 14:38:50  2016-04-01 14:38:50  2016-04-01 00:00:00
5  2016-03-21 13:47:45  2016-04-06 09:45:21  2016-03-21 00:00:00
6  2016-03-20 17:55:21  2016-03-23 02:48:59  2016-03-20 00:00:00
7  2016-03-16 18:55:19  2016-04-07 03:17:32  2016-03-16 00:00:00
8  2016-03-22 16:51:34  2016-03-26 18:18:10  2016-03-22 00:00:00
9  2016-03-16 13:47:02  2016-04-06 10:46:35  2016-03-16 00:00:00

The useful information is date. So we could delete the specific time.

In [24]:
autos['date_crawled'] = autos['date_crawled'].str[:10]
autos['last_seen'] = autos['last_seen'].str[:10]
autos['ad_created'] = autos['ad_created'].str[:10]
autos[['date_crawled','last_seen','ad_created']].head(10)

date_crawled   last_seen  ad_created
0   2016-03-26  2016-04-06  2016-03-26
1   2016-04-04  2016-04-06  2016-04-04
2   2016-03-26  2016-04-06  2016-03-26
3   2016-03-12  2016-03-15  2016-03-12
4   2016-04-01  2016-04-01  2016-04-01
5   2016-03-21  2016-04-06  2016-03-21
6   2016-03-20  2016-03-23  2016-03-20
7   2016-03-16  2016-04-07  2016-03-16
8   2016-03-22  2016-03-26  2016-03-22
9   2016-03-16  2016-04-06  2016-03-16

In [25]:
autos['date_crawled'].value_counts(normalize = True, dropna = False).sort_index()

2016-03-05    0.025372
2016-03-06    0.014050
2016-03-07    0.036033
2016-03-08    0.033264
2016-03-09    0.033037
2016-03-10    0.032211
2016-03-11    0.032603
2016-03-12    0.036963
2016-03-13    0.015640
2016-03-14    0.036632
2016-03-15    0.034277
2016-03-16    0.029504
2016-03-17    0.031612
2016-03-18    0.012913
2016-03-19    0.034752
2016-03-20    0.037789
2016-03-21    0.037314
2016-03-22    0.032955
2016-03-23    0.032273
2016-03-24    0.029401
2016-03-25    0.031550
2016-03-26    0.032252
2016-03-27    0.031095
2016-03-28    0.034855
2016-03-29    0.034132
2016-03-30    0.033740
2016-03-31    0.031818
2016-04-01    0.033740
2016-04-02    0.035496
2016-04-03    0.038574
2016-04-04    0.036529
2016-04-05    0.013079
2016-04-06    0.003161
2016-04-07    0.001384
Name: date_crawled, dtype: float64

The site was crawled everyday from 3/5/2016 to 4/7/2016. Basically each day there were similar listings been crawled.

In [26]:
autos['last_seen'].value_counts(normalize = True, dropna = False).sort_index()

2016-03-05    0.001074
2016-03-06    0.004339
2016-03-07    0.005413
2016-03-08    0.007355
2016-03-09    0.009628
2016-03-10    0.010620
2016-03-11    0.012376
2016-03-12    0.023781
2016-03-13    0.008864
2016-03-14    0.012624
2016-03-15    0.015868
2016-03-16    0.016446
2016-03-17    0.028058
2016-03-18    0.007335
2016-03-19    0.015826
2016-03-20    0.020599
2016-03-21    0.020599
2016-03-22    0.021384
2016-03-23    0.018595
2016-03-24    0.019752
2016-03-25    0.019194
2016-03-26    0.016798
2016-03-27    0.015599
2016-03-28    0.020868
2016-03-29    0.022335
2016-03-30    0.024752
2016-03-31    0.023843
2016-04-01    0.022872
2016-04-02    0.024876
2016-04-03    0.025186
2016-04-04    0.024504
2016-04-05    0.125000
2016-04-06    0.221612
2016-04-07    0.132025
Name: last_seen, dtype: float64

The 'last_seen' date indicate when the car were sold and the ads were removed. It seemed on 4/5/2016, 4/6/2016 and 4/7/2016 there were ten times more ads removed. It is unlikely more cars were sold in these 3 particular days. The best explaination would be this is due to the end of the crawling period.

In [27]:
autos['ad_created'].value_counts(normalize = True, dropna = False).sort_index().head(38)

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
2015-12-30    0.000021
2016-01-03    0.000021
2016-01-07    0.000021
2016-01-10    0.000041
2016-01-13    0.000021
2016-01-14    0.000021
2016-01-16    0.000021
2016-01-22    0.000021
2016-01-27    0.000062
2016-01-29    0.000021
2016-02-01    0.000021
2016-02-02    0.000041
2016-02-05    0.000041
2016-02-07    0.000021
2016-02-08    0.000021
2016-02-09    0.000021
2016-02-11    0.000021
2016-02-12    0.000041
2016-02-14    0.000041
2016-02-16    0.000021
2016-02-17    0.000021
2016-02-18    0.000041
2016-02-19    0.000062
2016-02-20    0.000041
2016-02-21    0.000062
2016-02-22    0.000021
2016-02-23    0.000083
2016-02-24    0.000041
2016-02-25    0.000062
2016-02-26    0.000041
2016-02-27    0.000124
2016-02-28    0.000207
2016-02-29    0.000165
Name: ad_created, dtype: float64

In [28]:
autos['ad_created'].value_counts(normalize = True, dropna = False).sort_index().tail(38)

2016-03-01    0.000103
2016-03-02    0.000103
2016-03-03    0.000868
2016-03-04    0.001488
2016-03-05    0.022934
2016-03-06    0.015310
2016-03-07    0.034773
2016-03-08    0.033264
2016-03-09    0.033120
2016-03-10    0.031921
2016-03-11    0.032934
2016-03-12    0.036777
2016-03-13    0.017004
2016-03-14    0.035269
2016-03-15    0.034008
2016-03-16    0.030000
2016-03-17    0.031281
2016-03-18    0.013595
2016-03-19    0.033636
2016-03-20    0.037851
2016-03-21    0.037541
2016-03-22    0.032769
2016-03-23    0.032107
2016-03-24    0.029339
2016-03-25    0.031674
2016-03-26    0.032314
2016-03-27    0.031012
2016-03-28    0.034979
2016-03-29    0.034070
2016-03-30    0.033554
2016-03-31    0.031860
2016-04-01    0.033719
2016-04-02    0.035186
2016-04-03    0.038822
2016-04-04    0.036901
2016-04-05    0.011798
2016-04-06    0.003244
2016-04-07    0.001240
Name: ad_created, dtype: float64

We notice most of the ads were created within the last month.

Next let's have a look at the 'registration_year' column.

In [29]:
autos['registration_year'].describe()

count    48400.000000
mean      2004.774091
std         88.791195
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

We notice we have the earlist year 1000 and latest year 9999, which obviously did not make any sense.

It would be reasonable to narrow the registration year to 1900 - 2016. Let's have a look at how many cases fall outside the range 1900-2016.

In [30]:
autos[~(autos['registration_year'].between(1900,2016))].shape[0] / autos[autos['registration_year'].between(1900,2016)].shape[0]

0.04032327400911358

Only about 4% of the cases fall outside the range 1900 - 2016. Therefore we could delete those cases.

In [31]:
autos = autos[autos['registration_year'].between(1900,2016)]

In [32]:
autos['registration_year'].describe()

count    46524.000000
mean      2002.926081
std          7.156633
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: registration_year, dtype: float64

In [37]:
autos['registration_year'].value_counts(normalize = True, dropna = False).head(10)

2000    0.067191
2005    0.062914
1999    0.062054
2004    0.058013
2003    0.057927
2006    0.057411
2001    0.056573
2002    0.053284
1998    0.050533
2007    0.048878
Name: registration_year, dtype: float64

We notice most cars were registered within the last 20 years.

Next we can have a look at how many different brand we have and what are their proportions.

In [38]:
autos['brand'].value_counts(normalize = True, dropna = False)

volkswagen        0.211353
bmw               0.109986
opel              0.107321
mercedes_benz     0.096466
audi              0.086622
ford              0.069985
renault           0.047137
peugeot           0.029877
fiat              0.025686
seat              0.018249
skoda             0.016422
nissan            0.015304
mazda             0.015239
smart             0.014208
citroen           0.014036
toyota            0.012746
hyundai           0.010038
sonstige_autos    0.009672
volvo             0.009114
mini              0.008770
mitsubishi        0.008189
honda             0.007845
kia               0.007093
alfa_romeo        0.006642
porsche           0.006083
suzuki            0.005932
chevrolet         0.005696
chrysler          0.003525
dacia             0.002644
daihatsu          0.002515
jeep              0.002278
subaru            0.002128
land_rover        0.002106
saab              0.001655
jaguar            0.001569
daewoo            0.001483
trabant           0.001397
r

Let's have a look at the mean odometer mileage of the top 10 brands.

In [40]:
top_brands = autos['brand'].value_counts().head(10).index
brand_mean_mileage = {}
for brand in top_brands:
    brand_mean_mileage[brand] = autos[autos['brand'] == brand]['odometer_km'].mean()
brand_mean_mileage

{'volkswagen': 128720.63459778298,
 'bmw': 132557.16239984366,
 'opel': 129339.07470458643,
 'mercedes_benz': 130838.9037433155,
 'audi': 129208.43672456576,
 'ford': 124247.542997543,
 'renault': 128144.09484724123,
 'peugeot': 127122.30215827338,
 'fiat': 117066.94560669456,
 'seat': 121166.0777385159}

We noticed the mean mileages of the top 10 brands are consistent.

Let's have a look at the mean prices of the top 10 brands.

In [41]:
brand_mean_price = {}
for brand in top_brands:
    brand_mean_price[brand] = autos[autos['brand'] == brand]['price_$'].mean()
brand_mean_price

{'volkswagen': 5621.73355028984,
 'bmw': 8606.653117060778,
 'opel': 2992.5137192068896,
 'mercedes_benz': 8657.285427807486,
 'audi': 9362.169478908188,
 'ford': 4064.650184275184,
 'renault': 2483.889192886457,
 'peugeot': 3100.6928057553955,
 'fiat': 2818.456066945607,
 'seat': 4417.942285041225}

We notice the mean prices varies a lot between different top 10 brands. Fiat is the cheapest brand and Audi is the most expensive among the top 10 brands.

In [47]:
bmm_series = pd.Series(brand_mean_mileage)
bmp_series = pd.Series(brand_mean_price)
brand_dataframe = pd.DataFrame(bmm_series, columns = ['mean_mileage'])
brand_dataframe

mean_mileage
volkswagen     128720.634598
bmw            132557.162400
opel           129339.074705
mercedes_benz  130838.903743
audi           129208.436725
ford           124247.542998
renault        128144.094847
peugeot        127122.302158
fiat           117066.945607
seat           121166.077739

In [49]:
brand_dataframe['mean_price'] = bmp_series
brand_dataframe

mean_mileage   mean_price
volkswagen     128720.634598  5621.733550
bmw            132557.162400  8606.653117
opel           129339.074705  2992.513719
mercedes_benz  130838.903743  8657.285428
audi           129208.436725  9362.169479
ford           124247.542998  4064.650184
renault        128144.094847  2483.889193
peugeot        127122.302158  3100.692806
fiat           117066.945607  2818.456067
seat           121166.077739  4417.942285

The mean mileage for the top 10 brands did not vary a lot while the price varied a lot.